core/src/main/java/org/jdbi/v3/core/mapper/reflect/ReflectionMappers.java

```diff
@@ -115,6 +118,35 @@ public ReflectionMappers setCaseChange(UnaryOperator<String> caseChange) {
         return this;
     }
 
+    /**
+     * Set the strategy Jdbi uses for Java accessibility rules.
+     * The legacy default is to call {@code setAccessible(true)} in certain cases when we try to use a Constructor, Method, or Field.
+     * In the future, this default will be changed to a no-op, to better interact with the Java module system.
+     */
+    @Beta
+    public ReflectionMappers setAccessibleObjectStrategy(Consumer<AccessibleObject> makeAccessible) {
+        this.makeAccessible = makeAccessible;
+        return this;
+    }
+
+    /**
+     * Set the strategy Jdbi uses for Java accessibility rules to a no-op.
+     */
+    @Beta
+    public ReflectionMappers disableAccessibleObjectStrategy() {
+        this.makeAccessible = accessibleObject -> {};
+        return this;
+    }
+
+    /**
+     * Use the accessibility strategy to potentially make a reflective operation accessible.
+     */
+    @Beta
+    public <T extends AccessibleObject> T makeAccessible(T accessibleObject) {
```

Comment:
I always felt that the config classes should be config. Now they actually contain logic.